src/tools/kata-ctl/src/ops/volume_ops.rs

```diff
@@ -0,0 +1,290 @@
+    }
+
+    // This behavior of deserializing and serializing comes from
+    // https://github.com/kata-containers/kata-containers/blob/cd27ad144e1a111cb606015c5c9671431535e644/src/runtime/pkg/direct-volume/utils.go#L57-L79
+    // Assuming that this is for the purpose of validating the json schema.
+    let unserialized_mount_info: DirectVolumeMountInfo = serde_json::from_str(mount_info)?;
+
+    let mount_info_file_path = mount_info_dir_path.join(KATA_MOUNT_INFO_FILE_NAME);
+    let serialized_mount_info = serde_json::to_string(&unserialized_mount_info)?;
+    fs::write(mount_info_file_path, serialized_mount_info)?;
+    Ok(None)
+}
+
+// remove deletes the direct volume path including all the files inside it.
+pub fn remove(volume_path: &str) -> Result<Option<String>> {
+    let path = join_path(KATA_DIRECT_VOLUME_ROOT_PATH, volume_path)?;
+    // removes path and any children it contains.
+    fs::remove_dir_all(path)?;
+    Ok(None)
+}
+
+pub fn get_volume_mount_info(volume_path: &str) -> Result<DirectVolumeMountInfo> {
+    let mount_info_file_path =
+        join_path(KATA_DIRECT_VOLUME_ROOT_PATH, volume_path)?.join(KATA_MOUNT_INFO_FILE_NAME);
+    let mount_info_file = fs::read_to_string(mount_info_file_path)?;
+    let mount_info: DirectVolumeMountInfo = serde_json::from_str(&mount_info_file)?;
+    Ok(mount_info)
+}
+
+// get_sandbox_id_for_volume finds the id of the first sandbox found in the dir.
+// We expect a direct-assigned volume is associated with only a sandbox at a time.
+pub fn get_sandbox_id_for_volume(volume_path: &str) -> Result<String> {
+    let dir_path = join_path(KATA_DIRECT_VOLUME_ROOT_PATH, volume_path)?;
+    let paths = fs::read_dir(dir_path)?;
+    for path in paths {
+        let path = path?;
+        // compare with MOUNT_INFO_FILE_NAME
+        if path.file_name() != KATA_MOUNT_INFO_FILE_NAME {
+            let file_name = path.file_name();
+            // turn file_name into String and return it
+            let file_name = file_name.to_str().ok_or_else(|| {
+                anyhow!(
+                    "failed to convert file_name {:?} to string",
+                    file_name.to_string_lossy()
+                )
+            })?;
+            return Ok(String::from(file_name));
+        }
+    }
+    return Err(anyhow!("no sandbox found for {}", volume_path));
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use kata_types::mount::DirectVolumeMountInfo;
+    use std::{collections::HashMap, fs};
+    use tempfile::tempdir;
+    use test_utils::skip_if_not_root;
+
+    #[test]
+    fn test_path_escape() {
+        // L3BhdGgvdG8vdm9s is the golang output of
+        // b64.URLEncoding.EncodeToString([]byte("/path/to/vol"))
+        let path = String::from("/path/to/vol");
+        let path_bytes = path.as_bytes();
+        let got = base64::encode(path_bytes);
+        let expected = String::from("L3BhdGgvdG8vdm9s");
+        assert!(got.eq(&expected));
+    }
```

Comment:
I'm not really sure why this test is required as it's not testing Kata code...?

But what we could use is a `test_join_path()` function that would pass in various volume paths and validate that the base64-encoded result is as expected.

Reply:
This was added initially to verify that the rust port would behave exactly as it would in go, but since we have tests for join_path, this is indeed no longer needed. I'll remove them right away.